guild-factory/src/factory.rs

```diff
@@ -33,6 +33,28 @@ pub trait FactoryModule: crate::config::ConfigModule {
         self.max_active_guilds().set(max_active_guilds);
     }
 
+    #[only_owner]
+    #[endpoint(upgradeGuild)]
+    fn upgrade_guild(
+        &self,
+        guild_address: ManagedAddress,
+        source_contract_address: ManagedAddress,
+    ) {
+        let guild_id = self.guild_ids().get_id_non_zero(&guild_address);
+        self.require_known_guild(guild_id);
+
+        let gas_left = self.blockchain().get_gas_left();
+        let code_metadata = self.get_default_code_metadata();
```

Comment:
I think `gas_left` should be computed after `self.get_default_code_metadata` to be accurate.